### PR TITLE
Add tests for fs.promises utimes function #527

### PR DESCRIPTION
**Pull request by biskit1**
Fixes issue #527 
Tests if `utimes` function returns a promise and implements promise versions of existing tests.

### CODE REVIEW

File: tests/spec/fs.utimes.spec.js [hunk 1]
```diff
@@ -175,3 +175,186 @@ describe('fs.utimes', function() {
...
+    fs.writeFile('/testfile', '')
+      .then(function () {
+        fs.utimes('/testfile', atime, mtime)
+          .then(function () {
+            fs.stat('/testfile')
+              .then(function (stat) {
+                expect(stat.mtime).to.equal(mtime);
+              })
+              .catch(function (error) {
+                expect(error).not.to.exist;
```
Comment (biskit1) [inline tests/spec/fs.utimes.spec.js]: Wondering if the `expect(error)` and entire `.catch()` portion is necessary for testing? 
If I left it out, what would happen if `.then()` failed?

File: tests/spec/fs.utimes.spec.js [hunk 2]
```diff
@@ -242,7 +242,7 @@ describe('fs.promises.utimes', function () {
     var atime = Date.parse('1 Oct 2000 15:33:22');
     var mtime = Date.parse('30 Sep 2000 06:43:54');
 
-    fs.utimes(1, atime, mtime)
+    return fs.utimes(1, atime, mtime)
```
Comment (biskit1) [inline tests/spec/fs.utimes.spec.js]: Not sure why all the tests passed when I forgot the return statements, but now that I added them just this one test is failing

Comment (biskit1) [reply tests/spec/fs.utimes.spec.js]: Should have called `futimes`, that's why it failed.

Comment (biskit1) [reply tests/spec/fs.utimes.spec.js]: @humphd, why do all the tests pass when I leave out the `return` statements?

Comment (humphd) [reply tests/spec/fs.utimes.spec.js]: The test hits the end of the stack, and nothing failed, so it says, "OK, I guess this worked."  Meanwhile, your file system calls are running async, and haven't started (or finished) yet:

```js
it('...', function() {
  startSomethingAsync();
  //test will end right here, before startSomethingAsync can finish
});
```

However, when you `return` a promise (or with callbacks use `done`), you block the test from exiting until the task is complete (promise is resolved, or callback done() is executed).

Comment (biskit1) [reply tests/spec/fs.utimes.spec.js]: thank you!

**Pull request merged**